test_communication/CMakeLists.txt

```diff
@@ -136,7 +136,7 @@ if(BUILD_TESTING)
     endforeach()
 
     # TODO different vendors can't talk to each other right now
-    if("${rmw_implementation1} " STREQUAL "${rmw_implementation2} " OR ("${rmw_implementation1} " STREQUAL "rmw_connext_cpp " AND "${rmw_implementation2} " STREQUAL "rmw_connext_dynamic_cpp ") OR ("${rmw_implementation1} " STREQUAL "rmw_connext_dynamic_cpp " AND "${rmw_implementation2} " STREQUAL "rmw_connext_cpp "))
+    if(rmw_implementation1 STREQUAL rmw_implementation2 OR (rmw_implementation1 STREQUAL "rmw_connext_cpp" AND rmw_implementation2 STREQUAL "rmw_connext_dynamic_cpp") OR (rmw_implementation1 STREQUAL "rmw_connext_dynamic_cpp" AND rmw_implementation2 STREQUAL "rmw_connext_cpp"))
```

Comment:
I am wondering why the linter isn't complaining about it in the first place (or if this file isn't being linted at all?).